### PR TITLE
feat: add variables for dependency ordering

### PR DESCRIPTION
**Pull request by morgante**
Also add an example + test for dependency control.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -47,6 +47,14 @@ locals {
 
 }
 
+resource "null_resource" "module_depends_on" {
+  count = length(var.module_depends_on) > 0 ? 1 : 0
+
+  triggers = {
+    value = length(var.module_depends_on)
```
Comment (onetwopunch) [inline main.tf]: Maybe I'm missing something, but how would this wait for _all_ the dependencies in the list? Seems the `decompress` resource is waiting on `module_depends_on` which is triggered once assuming the variable has a length of 1 or more. Can we update the test to show that this works for a list of more than one dependency?

Comment (morgante) [reply main.tf]: The `triggers` part depends on the *length* of the list, which causes it to be dependent on every resource within the list.

Comment (onetwopunch) [reply main.tf]: Ok cool that makes sense. But what happens if the list length stays the same but the contents of the list are different? Say I accidentally add a dependency on foo when I should have done it on bar so I change after the apply. Wouldn't the trigger not execute? Or is that not a use case we're trying to solve for?

Comment (morgante) [reply main.tf]: I believe it's based on the actual values in the list, regardless I don't think it's a scenario we need to solve for necessarily.

Comment (onetwopunch) [reply main.tf]: Make sense. Thanks for adding the test to be sure.

**Pull request merged**